python/pyspark/sql/session.py

```diff
@@ -252,6 +252,16 @@ def newSession(self):
         """
         return self.__class__(self._sc, self._jsparkSession.newSession())
 
+    @since(2.4)
+    def getActiveSession(self):
+        """
+        Returns the active SparkSession for the current thread, returned by the builder.
+        >>> s = spark.getActiveSession()
+        >>> spark._jsparkSession.getDefaultSession().get().equals(s.get())
+        True
+        """
+        return self._jsparkSession.getActiveSession()
```

Comment:
Does this return JVM instance?

Reply:
@HyukjinKwon Sorry for the late reply. Yes, this returns a JVM instance. 
In the scala code, ```SparkSession.getActiveSession``` returns an ```Option[SparkSession]```
I am not sure how to do a python equivalent of Scala's ```Option```. In the following code, is there a way to wrap the python ```session``` in else path to something equivalent of Scala's ```Option```?  If not, can I just return the python ```session```?
```
        if self._jsparkSession.getActiveSession() is None:
            return None
        else:
            return self.__class__(self._sc, self._jsparkSession.getActiveSession().get())
```

Reply:
Yea, I think we should return Python session one. JVM instance should not be exposed .. I assume returning `None` is fine. The thing is, we have the lack of session supports in PySpark. It's partially implemented but not very well tested as far as I can tell.

Can you add a set of tests for it, and manually test them as well? Actually, my guys say this is quite a big deal

Reply:
Yea, I think we should return Python session one. JVM instance should not be exposed .. I assume returning `None` is fine. The thing is, we have the lack of session supports in PySpark. It's partially implemented but not very well tested as far as I can tell.

Can you add a set of tests for it, and manually test them as well? Actually, my guts say this is quite a big deal

Reply:
@HyukjinKwon I add a set of tests. Some of them are borrowed from ```SparkSessionBuilderSuite.scala```